tokio-postgres/src/client.rs

```diff
@@ -450,6 +450,15 @@ impl Client {
         self.simple_query_raw(query).await?.try_collect().await
     }
 
+    /// Validates connection, timing out after specified duration.
+    pub async fn is_valid(&self, timeout: Duration) -> Result<(), Error> {
```

Comment:
Since timeouts are so easy in the async world, I'd prefer to only expose this API in the blocking client.

Reply:
Does Rust allow a module to expose functions to the workspace but not beyond? Otherwise, we would have to introduce `async` in the `postgres` crate, right?

Reply:
It does not, but that's not a problem. There's already async code in the postgres crate in the block_on calls.